### DISCUSSION TITLE
Is it possible to pass the variant prop to inputs currently?

### DISCUSSION THREAD

**Discussion by timayy** [vip-git/react-jsonschema-form-material-ui#381] [Q&A] [answered]
Material UI text fields have a [variant prop](https://material-ui.com/api/text-field/) that changes the way look - i.e. filled, outlined, standard.

Is it possible to pass this to inputs via the uiSchema? If not, is it possible to wrap the schema generated components in a ThemeProvider that just assigned a prop like variant?

Thanks!

**Comment by vip-git** [comment, marked as answer]
Hi @timayy ,

Yes you can pass any prop mentioned to change the text fields. 

There are 2 options:
1. Use `mui:variant` attribute in uiSchema 
 Example can be [seen here](https://github.com/vip-git/react-jsonschema-form-material-ui/blob/7e0e0f3ad73e91e46e46dbbfcea4399b0d6d094e/src/demo/examples/single/ui-schema.json#L2) for multiline .
```
you can prefix mui: followed by the prop name and value to use them in your uiSchema
```

2. You can make use of Theme and change the props directly for all inputs
    Example can be [seen here](https://github.com/vip-git/react-jsonschema-form-material-ui/blob/7e0e0f3ad73e91e46e46dbbfcea4399b0d6d094e/src/demo/theme.js#L19)
    
You can also read more about overriding attributes using theme from [Material UI Docs](https://material-ui.com/customization/components/#global-theme-override).
    
My recommendation would be to go 2nd way to be consistent and use theme overrides. 

I hope this answers your question @timayy.  Thanks for raising it.

You can also read more in this issue #361 where uiSchema was used but in your case theme override would be better option.

**Comment by timayy** [comment]
Thanks for the thorough answer, @vip-git. I appreciate it.

I know this isn't the place to ask, but before I make an issue, I thought I'd quick ask you -

I can't for the life of me get this package to work; both the latest main or next package. I've matched all the dependencies in a `create react app` project I have up, and whenever I import this package (`import MaterialJsonSchemaForm from "react-jsonschema-form-material-ui";`), my browser will hang, and eventually get the error: 

```
TypeError: Cannot read property 'call' of undefined
__webpack_require__
../my_app/frontend/create-react-app/node_modules/react-jsonschema-form-material-ui/dist/bundle.js:2
```

This code is identical to your [Basic Example Usage](https://github.com/vip-git/react-jsonschema-form-material-ui#basic-example-usage). But this error happens from just simply importing the package anywhere.

Identical error if using `react-jsonschema-form-material-ui@3.0.0-alpha.6` or `react-jsonschema-form-material-ui@3.0.0-alpha.6-mui-5`.

Only think I can think of is react create app uses an older webpack version compared to what you have in your devDependencies. 
Can you think of any reason why this happens?

**Reply by timayy** [reply]
@vip-git, having played around with this, trying to make it work with my create react app - I think it just outright doesn't work with CRA.

**Reply by vip-git** [reply]
Hi @timayy - Did you try this with version `2.2.2` ?

Version 3 is still being finalised. I will get back to you on this one by having a small boilerplate working code with CRA.

Thanks - V

**Reply by timayy** [reply]
I did get it working with `2.2.2`, but it's not ideal as there's a lot of `UNSAFE_componentWillMount` errors and I have to bring `react` and `react-dom` down to a low 16 version to get around a hook/state error.

Sorry about the bad news, @vip-git! Looking forward to version `3.0+` though! Hopefully you can get it working with react create app packages too; but I'm not sure if the later version of webpack you're using will make that tricky.

Thank you for getting back to me so quick, and the offer of a CRA boilerplate.

Appreciate your efforts.

**Reply by vip-git** [reply]
Thanks @timayy - I think there is soon going to be native support for [webpack 5](https://github.com/facebook/create-react-app/issues/9994) in CRA so that might resolve the issue as well.

I will try to get some boilerplate ready as well in any case so you have a reference .

**Reply by timayy** [reply]
Hey @vip-git, seen you've been super busy lately, awesome work!

How did you go with the web-pack dependencies?

**Reply by vip-git** [reply]
Hi @timayy - i am currently finalising `3.0.0` release part of that release i am creating the boilerplate for CRA. Expect that to be there by next week 23rd May - I will let you know if its done earlier. But when `3.0.0` finalised it will cover all aspects.

**Reply by vip-git** [reply]
Hi @timayy - Sorry for the delay in coming back with the boilerplate code.

I have created a repo now with [boilerplate code](https://github.com/vip-git/cra-react-jsonschema-form-material-ui-example) and deployed [output url](https://cra-react-jsonschema-form-material-ui-example.vercel.app)

I have released a new version [3.0.0-alpha.11](https://github.com/vip-git/react-jsonschema-form-material-ui/releases/tag/3.0.0-alpha.11) which now bundles using `rollup` to avoid webpack 5 conflicts with CRA.

i hope this helps in resolving your issue. Thanks for checking out react-jsonschema-form-material-ui. 

If you find similar issues please let me know.

Thanks -  V

**Reply by timayy** [reply]
Hey @vip-git, 

I apologize about the delay in response! Been a crazy few weeks!

Really, really appreciate you doing this! I had my own RJSF wrapper for MUI, but your components work much better - especially the multi-step stuff. Sorry if it was a hassle getting around the webpack 5 dependency, I know wrestling module bundlers can be painful.

I'll test out the boilerplate app, and if all is well I'll report back here, and begin replacing my RJSF system with react-jsonschema-form-material-ui.

Big thanks again, Vip - appreciate your efforts!
